mongo/datadog_checks/mongo/utils.py

```diff
@@ -27,7 +27,13 @@ def build_connection_string(hosts, scheme, username=None, password=None, databas
         netloc = host
 
     path_params = ""
-    query = urlencode(options or {})
+    url_options = dict(options)
+    if "username" in url_options:
+        url_options.pop("username")
+    if "password" in url_options:
+        url_options.pop("password")
+
+    query = urlencode(url_options or {})
```

Comment:
Just so I'm understanding this correctly, if `username` and `password` are passed as options, they'll be used in `datadog_checks.mongo.api` to initialize the client ([here](https://github.com/DataDog/integrations-core/blob/1cc20aa307831518a944c2f67c1c4f909961721c/mongo/datadog_checks/mongo/api.py#L39) I think?), but then we we want to remove `username` and `password` from the options when making the connection string?

Reply:
Yes!